src/services/reportCache.js

```diff
@@ -95,7 +95,7 @@ const cacheTopics = async (activityReportObjectiveId, topics = []) => {
   ]);
 };
 
-const cacheObjectiveMetadata = async (objective, reportId, metadata) => {
+const cacheObjectiveMetadata = async (objective, reportId, metadata, index) => {
```

Comment:
I think it makes sense to include the index in the metadata parameter instead of adding another (also I would personally prefer if it was called "order")